lib/handover.js

```diff
@@ -68,58 +72,52 @@ class Plugin {
         return res;
 
       let hasEnsReferral = false;
-      // Check NS records for referals to TLDs `.eth` and `._eth`
+      // Check NS records for referrals to TLDs `.eth` and `._eth`
       for (const rr of res.authority) {
         if (rr.type !== wire.types.NS)
           continue;
 
+        const ending = util.label(rr.data.ns, util.split(rr.data.ns), -1);
+
         // Look up the ENS resolver specified in the NS record
         // and query it for the user's original request
-        if (rr.data.ns.slice(-5) === '.eth.') {
+        if (ending === '_eth' || ending === 'eth') {
           hasEnsReferral = true;
 
           // If the recursive is being minimal, don't look up the name.
           // Send the SOA back and get the full query from the recursive .
           if (labels.length < 2) {
             return this.sendSOA();
           }
-
           this.logger.debug(
-            'Intercepted referral to .eth: %s %s -> NS: %s',
+            'Intercepted referral to .%s: %s %s -> %s NS: %s',
+            ending,
             name,
             wire.typesByVal[type],
+            rr.name,
             rr.data.ns
           );
-          data = await this.ethereum.resolveDnsFromEns(
-            name,
-            type,
-            rr.data.ns
-          );
-        }
-
-        // Look up an alternate (forked) ENS contract by the Ethereum
-        // address specified in the NS record, and query it for
-        // the user's original request
-        if (rr.data.ns.slice(-6) === '._eth.') {
-          hasEnsReferral = true;
 
-          // If the recursive is being minimal, don't look up the name.
-          // Send the SOA back and get the full query from the recursive .
-          if (labels.length < 2) {
-            return this.sendSOA();
+          try {
+            if (ending === 'eth') {
+              data = await this.ethereum.resolveDnsFromEns(
+                name,
+                type,
+                rr.data.ns
+              );
+            } else {
```

Comment:
Style-wise / readability wise down here I wonder if you should just add a comment that mentions `._eth` or even switch this if/else to a switch statement. I know we wouldnt be in this block if the tld wasnt one or the other but we might as well start preparing for things like #2 (`.sia`) and `.bit` and whatever else. So yeah maybe use a switch here so each TLD is explciit. and then in the future as we add more TLDs to this the user can configure a list of supported extensions and that list wil be checked up above on line 84... wha chu think?

Reply:
Yup I updated it to use switch/case and the comments to prepare for .sia, .bit ... etc